src/scipp/core/data_group.py

```diff
@@ -0,0 +1,192 @@
+            elif isinstance(item, (Variable, DataArray, Dataset)):
+                html = ''
+            else:
+                html = str(item)
+            out += f"<details style=\"padding-left:2em\"><summary>"\
+                   f"{name}: {_summarize(item)}</summary>{html}</details>"
+        return out
+
+    def _repr_html_(self):
+        out = ''
+        out += f"<details open=\"open\"><summary>DataGroup"\
+               f"({len(self)})</summary>"
+        out += self._make_html()
+        out += "</details>"
+        return out
+
+    def __repr__(self):
+        r = 'DataGroup(\n'
+        for name, var in self.items():
+            r += f'    {name}: {_summarize(var)}\n'
+        r += ')'
+        return r
+
+    def __eq__(self, other):
+        return _data_group_binary(operator.eq, self, other)
+
+    def __add__(self, other):
+        return _data_group_binary(operator.add, self, other)
+
+    def __mul__(self, other):
+        return _data_group_binary(operator.mul, self, other)
+
+    def _call_method(self, func: Callable) -> DataGroup:
+        """Call method on all values and return new DataGroup containing the results."""
+        return DataGroup({key: func(value) for key, value in self.items()})
+
+    @property
+    def bins(self):
+        # TODO Returning a regular DataGroup here may be wrong, since the `bins`
+        # property provides a different set of attrs and methods.
+        return self._call_method(operator.attrgetter('bins'))
+
+    def copy(self, *args, **kwargs):
+        return self._call_method(operator.methodcaller('copy', *args, **kwargs))
+
+    def bin(self, *args, **kwargs):
+        return self._call_method(operator.methodcaller('bin', *args, **kwargs))
+
+    def group(self, *args, **kwargs):
+        return self._call_method(operator.methodcaller('group', *args, **kwargs))
+
+    def groupby(self, *args, **kwargs):
+        return self._call_method(operator.methodcaller('groupby', *args, **kwargs))
+
+    def hist(self, *args, **kwargs):
+        return self._call_method(operator.methodcaller('hist', *args, **kwargs))
+
+    def sum(self, *args, **kwargs):
+        return self._call_method(operator.methodcaller('sum', *args, **kwargs))
+
+    def mean(self, *args, **kwargs):
+        return self._call_method(operator.methodcaller('mean', *args, **kwargs))
+
+    def min(self, *args, **kwargs):
+        return self._call_method(operator.methodcaller('min', *args, **kwargs))
+
+    def max(self, *args, **kwargs):
+        return self._call_method(operator.methodcaller('max', *args, **kwargs))
+
+    def transform_coords(self, *args, **kwargs):
+        return self._call_method(
+            operator.methodcaller('transform_coords', *args, **kwargs))
+
+    def to(self, *args, **kwargs):
+        return self._call_method(operator.methodcaller('to', *args, **kwargs))
+
+    def plot(self, *args, **kwargs):
+        import plopp
+        return plopp.plot(self, *args, **kwargs)
```

Comment:
Not sure what will happen when you have nested groups. I think `plopp` only works with a flat dict.

Reply:
I think it will raise?